From d1078877a39d47d9626cedfc0417c25240a76e86 Mon Sep 17 00:00:00 2001
From: Abhishek <apatgaonkar@habana.ai>
Date: Mon, 30 Sep 2024 12:10:05 +0300
Subject: [PATCH] [FSW-419] fs1: rma: remove rma manifest usage

As per discussion with security spec, rma manifest bit
indicating if manifest can be loaded in rma is no
longer used. Only rma efuse bit is used to decide
if manifest can be loaded with lower svn and revoked
key. This patch removes all relevant code.

Change-Id: I3d46b96e418d3f9946cc4cca42e8e6028f3ad984
---

diff --git a/common/common.c b/common/common.c
index fe34d97..b232389 100644
--- a/common/common.c
+++ b/common/common.c
@@ -161,7 +161,7 @@
 	ret = secure_boot_verify_image_signature((uint8_t *)boot_ptr_addr, img_plus_key_len,
 			(uint8_t *)(boot_ptr_addr + img_plus_key_len - boot_ptr->pub_key_sz),
 			boot_ptr->pub_key_sz, (uint8_t *)(boot_ptr_addr + img_plus_key_len), boot_ptr->sig_sz,
-			boot_ptr->sec_ver, boot_ptr->key_idx, boot_ptr->rma_manifest);
+			boot_ptr->sec_ver, boot_ptr->key_idx);
 
 	if (ret == SECUREBOOT_VERIFY_SUCCESS) {
 		boot_rom_status_update_bringup(boot_desc_verif_base, PROGRESS_VERIFY_SUCCESS);
diff --git a/common/secure_boot.c b/common/secure_boot.c
index 9f00e53..c69e9f7 100644
--- a/common/secure_boot.c
+++ b/common/secure_boot.c
@@ -39,7 +39,7 @@
 
 int secure_boot_verify_image_signature(const uint8_t *image, uint32_t image_len, uint8_t *key,
 				uint32_t key_len, uint8_t *sig, uint32_t sig_len,
-				uint8_t sec_ver, uint8_t key_index, uint8_t rma_manifest)
+				uint8_t sec_ver, uint8_t key_index)
 {
 	int ret;
 
@@ -49,11 +49,10 @@
 	if ((key_len > TOTAL_PUB_KEY_PROPS_SZ) || (sig_len > SIG_SIZE))
 		return SECUREBOOT_VERIFY_FAILED;
 
-	if (security_check_valid_fw_version(sec_ver, rma_manifest) != SECURITY_FW_VERSION_SECURE) {
+	if (security_check_valid_fw_version(sec_ver) != SECURITY_FW_VERSION_SECURE)
 		return SECUREBOOT_VERIFY_VER_FAILED;
-	}
 
-	ret = security_check_key_revoke_status(key_index, rma_manifest);
+	ret = security_check_key_revoke_status(key_index);
 	if (ret != 1)
 		return SECUREBOOT_VERIFY_KEY_REVOKED;
 
diff --git a/common/secure_boot.h b/common/secure_boot.h
index 29010cb..5a8243b 100644
--- a/common/secure_boot.h
+++ b/common/secure_boot.h
@@ -15,7 +15,7 @@
 int secure_boot_verify_image_hash384(uint8_t *exp_hash, unsigned char *img, uint32_t img_len);
 int secure_boot_verify_image_signature(const uint8_t *image, uint32_t image_len, uint8_t *key,
 				uint32_t key_len, uint8_t *sig, uint32_t sig_len,
-				uint8_t sec_ver, uint8_t key_index, uint8_t rma_manifest);
+				uint8_t sec_ver, uint8_t key_index);
 bool boot_rom_verify_fw_image_hash384(enum hl_fw_manifest_type manifest_idx,
 	enum hl_fw_img_type img_type);
 
diff --git a/common/security.c b/common/security.c
index ebab361..b4d00f9 100644
--- a/common/security.c
+++ b/common/security.c
@@ -10,9 +10,9 @@
 #include <string.h>
 #include <security.h>
 
-int security_check_valid_fw_version(uint32_t fw_version, uint8_t rma_manifest)
+int security_check_valid_fw_version(uint32_t fw_version)
 {
-	if (!(!!boot_rom_global.sec_data.rma_enabled && !!rma_manifest)) {
+	if (!boot_rom_global.sec_data.rma_enabled) {
 		if ((fw_version < boot_rom_global.sec_data.sec_ver) ||
 			(fw_version > SECURITY_MAX_FW_SEC_VERSION)) {
 			return SECURITY_FW_VERSION_NOT_SECURE;
@@ -22,7 +22,7 @@
 	return SECURITY_FW_VERSION_SECURE;
 }
 
-int security_check_key_revoke_status(uint32_t key_index, uint8_t rma_manifest)
+int security_check_key_revoke_status(uint32_t key_index)
 {
 	uint8_t key_revoke;
 
@@ -33,8 +33,7 @@
 
 	/* if the key is not revoked, return 1 */
 	if (((key_revoke & (1 << key_index)) == 0) ||
-		(!!boot_rom_global.sec_data.rma_enabled &&
-		!!rma_manifest))
+		!!boot_rom_global.sec_data.rma_enabled)
 		return 1;
 
 	/* no valid key found */
diff --git a/common/security.h b/common/security.h
index e0213a0..1d03c35 100644
--- a/common/security.h
+++ b/common/security.h
@@ -11,8 +11,8 @@
 #define SECURITY_MAX_FW_SEC_VERSION		128
 
 void boot_rom_security_init(void);
-int security_check_key_revoke_status(uint32_t key_index, uint8_t rma_manifest);
+int security_check_key_revoke_status(uint32_t key_index);
 int security_rom_get_key_hash_by_idx(uint32_t *hash, uint8_t key_index);
-int security_check_valid_fw_version(uint32_t fw_version, uint8_t rma_manifest);
+int security_check_valid_fw_version(uint32_t fw_version);
 
 #endif /* _SECURITY_H_ */
